modules/sievefilters/functions.php

```diff
@@ -284,6 +284,22 @@ function get_sieve_client_factory($site_config)
     }
 }
 
+if (!hm_exists('encode_or_decode_blocked_senders_script')) {
+    function encode_or_decode_blocked_senders_script($script, $index = 1, $action = "decode")
```

Comment:
That's better. Can you just rename the method to be more consistent with its usage? Something like prepare_sieve_script and action could be read/write or direction=straight/reverse or something more obvious. encode and decode is not good here as it has twofold meaning - base64 decoding but json encoding or base64 encoding but json decoding.